app/ht_indexer/tests/document_retriever_service_tests/full_text_search_retriever_service_test.py

```diff
@@ -58,51 +65,67 @@ def test_generate_metadata(self, get_item_metadata):
-                "password": "guest",
-                "host": "get_rabbit_mq_host_name",
-                "queue_name": "test_producer_queue",
-                "requeue_message": False,
-                "query_field": "item",
-                "batch_size": 1,
-            }
-        ],
-        indirect=["retriever_parameters"],
-    )
-    def test_something(self, retriever_parameters):
-        # retriever_parameters["host"] will be the value from the fixture
-        assert retriever_parameters["host"] == "rabbitmq"
-
-    @pytest.mark.parametrize("retriever_parameters", [{"user": "guest", "password": "guest",
-                                                       "host": "get_rabbit_mq_host_name",
-                                                       "queue_name": "test_producer_queue",
-                                                       "requeue_message": False,
-                                                       "query_field": "item",
-                                                       "batch_size": 1}],
-                             indirect=["retriever_parameters"])
-    def test_full_text_search_retriever_service(self, retriever_parameters, get_document_retriever_service,
-                                                consumer_instance):
+    def test_full_text_search_retriever_service(self, get_rabbit_mq_host_name,
+                                                get_retriever_service_solr_parameters,
+                                                solr_catalog_url):
+
         """ Use case: Check if the message is sent to the queue"""
+
+        queue_name = "test_full_text_search_retriever_service"
+        queue_user = "guest"
+        queue_pass = "guest"
+        # Define the consumer instance
+        consumer_instance = QueueConsumer(
+            queue_user,
+            queue_pass,
+            get_rabbit_mq_host_name,
+            queue_name,
+            False,
+            1
+        )
+
         # Clean up the queue
-        consumer_instance.conn.ht_channel.queue_purge(consumer_instance.queue_name)
+        consumer_instance.ht_channel.queue_purge(consumer_instance.queue_name)
 
         list_documents = ['nyp.33433082002258', 'not_exist_document']
 
-        get_document_retriever_service.full_text_search_retriever_service(
+        document_retriever_service_obj = FullTextSearchRetrieverQueueService(
+            queue_name,
+            get_rabbit_mq_host_name,
+            queue_user,
+            queue_pass,
+            solr_catalog_url,
+            "solr_user",
+            "solr_password",
+            get_retriever_service_solr_parameters
+        )
+
+        # Service to push a message
+        document_retriever_service_obj.full_text_search_retriever_service(
             list_documents,
-            retriever_parameters["query_field"]
+    "item"
         )
 
-        assert 1 == consumer_instance.conn.get_total_messages()
+        time.sleep(1) # Give RabbitMQ time to register the message
```

Comment:
My expectation is that `consume_message` would wait until there was a message on the queue (up to `inactivity_timeout`), so we shouldn't need the sleep here. If it seems to be required, I would want to again look a bit at the exact sequence of what was happening.